pkg/liquidity-mining/contracts/VotingEscrowRemapper.sol

```diff
@@ -0,0 +1,286 @@
+     * @param localUser - Address of the user on the local chain which are querying the remote address for.
+     * @param chainId - The chain ID of the network which the remote user is on.
+     */
+    function getRemoteUser(address localUser, uint256 chainId) public view returns (address) {
+        return _localToRemoteAddressMap[chainId][localUser];
+    }
+
+    function getRemappingManager(address localUser) public view returns (address) {
+        return _localRemappingManager[localUser];
+    }
+
+    function setOmniVotingEscrow(IOmniVotingEscrow omniVotingEscrow) external authenticate {
+        _omniVotingEscrow = omniVotingEscrow;
+        emit OmniVotingEscrowUpdated(omniVotingEscrow);
+    }
+
+    // Remapping Setters
+
+    /**
+     * @notice Sets up a mapping from `localUser`'s veBAL balance to `remoteUser` for chain `chainId`.
+     * @dev In order to set up a remapping on this contract, `localUser` must be a smart contract which has been
+     * allowlisted to hold veBAL. EOAs are expected to set up any delegation of their veBAL on L2s directly.
+     * @param localUser - The address of a contract allowlisted on the `SmartWalletChecker`.
+     * @param remoteUser - The address to receive `localUser`'s balance of veBAL on the remote chain.
+     * @param chainId - The chain id of the remote chain on which `remoteUser` resides.
+     */
+    function setNetworkRemapping(
+        address localUser,
+        address remoteUser,
+        uint256 chainId
+    ) external payable {
+        _require(msg.sender == localUser || msg.sender == _localRemappingManager[localUser], Errors.SENDER_NOT_ALLOWED);
+        require(_isAllowedContract(localUser), "Only contracts which can hold veBAL can set up a mapping");
+        require(remoteUser != address(0), "Zero address cannot be used as remote user");
+        IOmniVotingEscrow omniVotingEscrow = getOmniVotingEscrow();
+        require(address(omniVotingEscrow) != address(0), "Omni voting escrow not set");
+
+        // Since we keep a 1-to-1 local-remote mapping for each chain, the remote address must not
+        // already be in use by another local user.
+        // Note that this means that it is possible to frontrun this call to grief a user by taking up their
+        // selected remote address. This is mitigated somewhat by restricting potential attackers to the
+        // set of contracts that are allowlisted to hold veBAL (and their remapping managers). Should
+        // one of them grief, then Balancer governance can remove them from these allowlists.
+        require(
+            _remoteToLocalAddressMap[chainId][remoteUser] == address(0),
+            "Cannot overwrite an existing mapping by another user"
+        );
+
+        // TODO: confirm this.
+        // require(
+        //     _localToRemoteAddressMap[chainId][remoteUser] == address(0),
+        //     "Cannot to remap to an address that is in use locally"
+        // );
```

Comment:
@nventuro I think we should add this restriction.

Whenever a mapping takes place, 
- You can't step somebody else's remote address
- The remote address in the local chain shouldn't be remapped, since it 'belongs' to the local user.

On the other hand, we'll also be adding a check for the veBAL balance for the remote address.
Something like this:
```
require(veBAL.balanceOf(remoteUser) == 0)
```